src/core/net/tcp6.hpp

```diff
@@ -0,0 +1,302 @@
+    {
+        friend class Tcp;
+        friend class LinkedList<Listener>;
+    };
+
+    /**
+     * This class implements TCP header parsing.
+     *
+     */
+    OT_TOOL_PACKED_BEGIN
+    class Header
+    {
+    public:
+        enum : uint8_t
+        {
+            kChecksumFieldOffset = 16, ///< The byte offset of the Checksum field in the TCP header.
+        };
+
+        /**
+         * This method returns the TCP Source Port.
+         *
+         * @returns The TCP Source Port.
+         *
+         */
+        uint16_t GetSourcePort(void) const { return HostSwap16(mSource); }
+
+        /**
+         * This method returns the TCP Destination Port.
+         *
+         * @returns The TCP Destination Port.
+         *
+         */
+        uint16_t GetDestinationPort(void) const { return HostSwap16(mDestination); }
+
+        /**
+         * This method returns the TCP Sequence Number.
+         *
+         * @returns The TCP Sequence Number.
+         *
+         */
+        uint32_t GetSequenceNumber(void) const { return HostSwap32(mSequenceNumber); }
+
+        /**
+         * This method returns the TCP Acknowledgment Sequence Number.
+         *
+         * @returns The TCP Acknowledgment Sequence Number.
+         *
+         */
+        uint32_t GetAcknowledgmentNumber(void) const { return HostSwap32(mAckNumber); }
+
+        /**
+         * This method returns the TCP Flags.
+         *
+         * @returns The TCP Flags.
+         *
+         */
+        uint16_t GetFlags(void) const { return HostSwap16(mFlags); }
+
+        /**
+         * This method returns the TCP Window.
+         *
+         * @returns The TCP Window.
+         *
+         */
+        uint16_t GetWindow(void) const { return HostSwap16(mWindow); }
+
+        /**
+         * This method returns the TCP Checksum.
+         *
+         * @returns The TCP Checksum.
+         *
+         */
+        uint16_t GetChecksum(void) const { return HostSwap16(mChecksum); }
+
+        /**
+         * This method returns the TCP Urgent Pointer.
+         *
+         * @returns The TCP Urgent Pointer.
+         *
+         */
+        uint16_t GetUrgentPointer(void) const { return HostSwap16(mUrgentPointer); }
+
+    private:
+        uint16_t mSource;
+        uint16_t mDestination;
+        uint32_t mSequenceNumber;
+        uint32_t mAckNumber;
+        uint16_t mFlags;
+        uint16_t mWindow;
+        uint16_t mChecksum;
+        uint16_t mUrgentPointer;
+
+    } OT_TOOL_PACKED_END;
+
+    /**
+     * This constructor initializes the object.
+     *
+     * @param[in] aInstance  A reference to the OpenThread instance.
+     *
+     */
+    explicit Tcp(Instance &aInstance);
+
+    /**
+     * See documentation for otTcpEndpointInitialize in include/openthread/tcp.h.
+     */
+    Error Initialize(Endpoint &aEndpoint, otTcpEndpointInitializeArgs &aArgs);
+
+    /**
+     * See documentation for otTcpEndpointGetInstance in include/openthread/tcp.h.
+     */
```

Comment:
I suggest we copy the documentation of methods/functions again here, instead of just pointing to `tcp.h`. 

Having it in the OT internal types again can be useful (someone reading the code and starts from internal definition. Also some IDEs (editors) can parse and show documentation when writing code which can be helpful).

In other modules, we usually repeat the documentation (we have in public OT header and then again in internal types/definitions).

Reply:
OK, I am happy to do this. My original reasoning was to avoid having duplicate comments (since then we would have to remember to change the other copy if we decide to change things). But if there's precedent for copying the comments elsewhere in OpenThread, then I agree that we should do that.